### PR TITLE
Change to value object

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
I have removed pecl and added zip too @franmomu

**Review by wbloszyk — APPROVED** [review]

File: config/projects.yaml [hunk 1]
```diff
@@ -224,16 +224,14 @@ doctrine-mongodb-admin-bundle:
     branches:
         master:
             php: ['7.3', '7.4', '8.0']
-            php_extensions: [mongodb-1.9.0]
-            tools: [pecl]
+            php_extensions: ['mongodb-1.9.0, zip']
```
Comment (jordisala1991) [inline config/projects.yaml:227]: ['mongodb-1.9.0', 'zip']

Comment (franmomu) [inline config/projects.yaml:227]:
```suggestion
            php_extensions: ['mongodb-1.9.0', 'zip']
```

Comment (VincentLanglet) [reply config/projects.yaml]: Fixed

Comment (VincentLanglet) [reply config/projects.yaml]: Fixed

**Review by wbloszyk — APPROVED** [review]

File: src/Domain/Value/Branch.php [hunk 2]
```diff
@@ -138,17 +142,20 @@ public function hasTool(string $toolName): bool
...
     }
 
-    public function hasPhpExtensions(string $phpExtensionName): bool
+    public function hasPhpExtension(string $phpExtensionName): bool
     {
         foreach ($this->phpExtensions() as $phpExtension) {
-            if ($phpExtension === $phpExtensionName || explode('-', $phpExtension)[0] === $phpExtensionName) {
+            if (
+                $phpExtension->toString() === $phpExtensionName
+                || explode('-', $phpExtension->toString())[0] === $phpExtensionName
```
Comment (phansys) [inline src/Domain/Value/Branch.php:157]:
```suggestion
                || explode('-', $phpExtension->toString(), 2)[0] === $phpExtensionName
```

**Review by phansys — APPROVED** [review]

File: src/Domain/Value/Branch.php [hunk 3]
```diff
@@ -45,10 +45,10 @@ final class Branch
     private PhpVersion $targetPhpVersion;
 
     /**
-     * @param array<string, PhpVersion> $phpVersions
-     * @param Tool[]                    $tools
-     * @param string[]                  $phpExtensions
-     * @param Variant[]                 $variants
+     * @param PhpVersion[]   $phpVersions
```
Comment (OskarStark) [inline src/Domain/Value/Branch.php:48]: Please keep the old version, we have string as key

File: src/Domain/Value/PhpExtension.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Domain\Value;
+
+/**
+ * @author Oskar Stark <oskarstark@googlemail.com>
```
Comment (OskarStark) [inline src/Domain/Value/PhpExtension.php:17]: 😄

File: src/Domain/Value/PhpVersion.php [hunk 5]
```diff
@@ -40,11 +30,6 @@ public static function fromString(string $version): self
         return new self($version);
     }
 
-    public function isStable(): bool
```
Comment (OskarStark) [inline src/Domain/Value/PhpVersion.php:43]: isn't this used? 🤔

Comment (VincentLanglet) [reply src/Domain/Value/PhpVersion.php:43]: No

Comment (OskarStark) [reply src/Domain/Value/PhpVersion.php:43]: Ok I thought we used it in the template, but maybe we can keep it when 8.1 or nightly will be tested

Comment (VincentLanglet) [reply src/Domain/Value/PhpVersion.php:43]: That way I would see if we could do this in another way.
Maybe isStable = not nightly, or isStable = not 8.1, or something else...

By the way is was written, we forgot to update the code to add 8.0 in the list of stable php.
And there was 7.0, 7.1, 7.2 which we don't support anymore.

**Review by OskarStark — APPROVED** [review]

**Pull request merged**